src/tests/ftest/deployment/target_failure.py

```diff
@@ -0,0 +1,417 @@
+        ior_results = {}
+        job_num = 1
+        oclass = self.params.get("oclass", "/run/object_class/wo_rf/*")
+        job = threading.Thread(
+            target=self.run_ior_report_error,
+            args=[ior_results, job_num, "test_file_1", oclass, self.pool, self.container])
+
+        job.start()
+
+        # Wait for a few seconds for IOR to start.
+        self.log.info("Waiting 5 sec for IOR to start writing data...")
+        time.sleep(5)
+
+        errors = []
+
+        # 3. Exclude one target while IOR is running.
+        self.pool.exclude(ranks=[1], tgt_idx="0")
+        self.measure_rebuild_time(operation="Exclude 1 target", pool=self.pool)
+
+        # Wait until the IOR thread ends.
+        job.join()
+
+        # 4. Verify that the IOR failed.
+        self.log.info("----- IOR results 1 -----")
+        self.log.info(ior_results)
+        if ior_results[job_num][0]:
+            ior_error = ior_results[job_num][-1]
+            errors.append(
+                "First IOR was supposed to fail, but worked! {}".format(ior_error))
+
+        # 5. Verify that the container's Health property is UNCLEAN.
+        if not self.check_container_health(
+                container=self.container, expected_health="UNCLEAN"):
+            errors.append("Container health isn't UNCLEAN after first IOR!")
+
+        # 6. Reintegrate the excluded target.
+        self.pool.reintegrate(rank="1", tgt_idx="0")
+        self.measure_rebuild_time(operation="Reintegrate 1 target", pool=self.pool)
+
+        # 7. Verify that the container's Health property is HEALTHY.
+        if not self.check_container_health(
+                container=self.container, expected_health="HEALTHY"):
+            errors.append("Container health isn't HEALTHY after reintegrate!")
+
+        # 8. Run IOR again.
+        ior_results = {}
+        self.run_ior_report_error(
+            job_num=job_num, results=ior_results, file_name="test_file_2", oclass=oclass,
+            pool=self.pool, container=self.container)
+
+        # 9. Verify that there's no error this time.
+        self.log.info("----- IOR results 2 -----")
+        self.log.info(ior_results)
+        if not ior_results[job_num][0]:
+            ior_error = ior_results[job_num][1]
+            errors.append("Error found in second IOR run! {}".format(ior_error))
+
+        report_errors(test=self, errors=errors)
+
+    def test_target_failure_with_rf(self):
+        """Jira ID: DAOS-10001.
+
+        Verify that failing (excluding) one target from each server rank would cause an
+        error to the ongoing IOR even if redundancy factor is used. Also verify that
+        reintegrating the excluded target will bring back the system to the usable state.
+
+        1. Run two server ranks and create a pool and a container with --properties=rf:1.
+        2. Run IOR with --dfs.oclass RP_2G1 --dfs.dir_oclass RP_2G1
```

Comment:
Not sure if it matters for this test, but for performance at least, we use `--dfs.oclass RP_2GX --dfs.dir_oclass RP_2G1`

Reply:
Okay. Trying with more object classes is one way to enhance the fault management tests.